classes/ActionScheduler_ListTable.php

```diff
@@ -0,0 +1,354 @@
+	 *
+	 * @param ActionScheduler_Action $item
+	 *
+	 * @return string
+	 */
+	protected function get_recurrence( $item ) {
+		$recurrence = $item->get_schedule();
+		if ( method_exists( $recurrence, 'interval_in_seconds' ) ) {
+			return self::human_interval( $recurrence->interval_in_seconds() );
+		}
+		return __( 'Non-repeating', 'action-scheduler' );
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	protected function get_items_query_limit() {
+		return $this->get_items_per_page( $this->package . '_items_per_page', $this->items_per_page );
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	protected function get_items_query_offset() {
+		global $wpdb;
+
+		$per_page = $this->get_items_query_limit();
+		$current_page = $this->get_pagenum();
+		if ( 1 < $current_page ) {
+			return $per_page * ( $current_page - 1 );
+		}
+
+		return 0;
+	}
+
+	/**
+	 * Serializes the argument of an action to render it in a human friendly format.
+	 *
+	 * @param array $row The array representation of the current row of the table
+	 *
+	 * @return string
+	 */
+	public function column_args( array $row ) {
+		return '<code>' . json_encode( $row['args'] ) . '</code>';
+	}
+
+	/**
+	 * Prints the comments, which are the log entries. It needs to be named comments otherwise it won't pickup
+	 * the WordPress styles to make the number pretty.
+	 *
+	 * It will also render all the log entries, but it will be hidden and rendered on click in the number as a modal.
+	 *
+	 * @param array $row Action array.
+	 */
+	public function column_comments( array $row ) {
+		echo '<div id="log-' . $row['ID'] . '" class="log-modal hidden" style="max-width:800px">';
+		echo '<h3>Log entries for ' . $row['ID'] . '</h3>';
+		foreach ( $row['comments'] as $log ) {
+			echo '<p>' . $log->get_message() . '</p>';
+		}
+		echo '</div>';
+
+		echo '<div class="post-com-count-wrapper">';
+		echo '<a href="#" data-log-id="' . $row['ID'] . '" class="post-com-count post-com-count-approved log-modal-open">';
+		echo '<span class="comment-count-approved">' . esc_html( count( $row['comments'] ) ) . '</span>';
+		echo '</a>';
+		echo '</div>';
+	}
+
+	/**
+	 * Prints the scheduled date in a human friendly format.
+	 *
+	 * @param array $row The array representation of the current row of the table
+	 */
+	public function column_scheduled( $row ) {
+		$next_timestamp = $row['scheduled']->next()->format( 'U' );
+		echo get_date_from_gmt( gmdate( 'Y-m-d H:i:s', $next_timestamp ), 'Y-m-d H:i:s' );
+		if ( gmdate( 'U' ) > $next_timestamp ) {
+			printf( __( ' (%s ago)', 'action-scheduler' ), human_time_diff( gmdate( 'U' ), $next_timestamp ) );
+		} else {
+			echo ' (' . human_time_diff( gmdate( 'U' ), $next_timestamp ) . ')';
+		}
+	}
+
+	/**
+	 * Returns if the current action is finished or pending.
+	 *
+	 * @param ActionScheduler_Action $item
+	 */
+	protected function get_status( $item ) {
+		if ( $item->is_finished() ) {
+			return 'Completed';
+		}
+
+		return 'Pending';
```

Comment:
These status strings should be passed through i18n functions so they are available for translation.